File: domain/src/main/java/org/openmedstack/eventstore/IStoreEvents.java
package org.openmedstack.eventstore;

import java.util.concurrent.CompletableFuture;

public interface IStoreEvents extends AutoCloseable
{
    //IPersistStreams advanced ();

    CompletableFuture<IEventStream> createStream(String bucketId, String streamId);

    CompletableFuture<IEventStream> openStream(String bucketId, String streamId, int minRevision, int maxRevision);

    CompletableFuture<IEventStream> openStream(Snapshot snapshot, int maxRevision);
}

File: domain/src/main/kotlin/org/openmedstack/eventstore/IStoreEvents.kt
package org.openmedstack.eventstore

import org.openmedstack.domain.Memento
import java.util.concurrent.CompletableFuture

interface IStoreEvents : AutoCloseable {
    fun createStream(bucketId: String?, streamId: String): CompletableFuture<IEventStream>
    fun openStream(bucketId: String?, streamId: String, minRevision: Int, maxRevision: Int): CompletableFuture<IEventStream>
    fun <TMemento : Memento> openStream(snapshot: Snapshot<TMemento>, maxRevision: Int): CompletableFuture<IEventStream>
}
